### PR TITLE
remove tight coupling to test execution

### PR DESCRIPTION
**Pull request by jjelschen**
some small refactorings to properly decouple test execution code from the rest. Prerequisite for taking out test execution into its own repository w/ separate packaging and releasing.

### CODE REVIEW

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/CancelledTestExecutionDetailsTest.xtend [hunk 1]
```diff
@@ -14,7 +14,7 @@ import javax.ws.rs.core.MediaType
 import org.eclipse.jgit.api.Git
 import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
 import org.junit.Test
-import org.testeditor.web.backend.persistence.AbstractPersistenceIntegrationTest
```
Comment (kbayraktar) [inline org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/CancelledTestExecutionDetailsTest.xtend]: Reconsider moving this file to persistence ?

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestSuiteExecutorIntegrationTest.xtend [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.testeditor.web.backend.testexecution
```
Comment (kbayraktar) [inline org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestSuiteExecutorIntegrationTest.xtend:1]: same same ... ;O)

**Review by kbayraktar — CHANGES_REQUESTED** [review]

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/CancelledTestExecutionDetailsTest.xtend (hunk 1 above)
Comment (jjelschen) [reply org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/CancelledTestExecutionDetailsTest.xtend]: This is actually an integration test that needs to be in the persistence. I renamed the class, accordingly.

The imports do look confusing in the diff, but are actually correct: the import from *.persistence is not necessary anymore, as this is the local package now, whereas dependencies on *testexecution now need to be imported, as they are not in the same package, anymore.

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestSuiteExecutorIntegrationTest.xtend (hunk 2 above)
Comment (jjelschen) [reply org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestSuiteExecutorIntegrationTest.xtend:1]: same reasoning, too :grin:

**Pull request merged**